server.js

```diff
@@ -1,88 +1,73 @@
 if (config.auth && config.auth.enabled) {
   let auth = null;
   if (config.auth.plugin === 'auth-marklogic-digest-basic') {
     auth = require("./src/koop/authMarkLogic")(config.auth.options);
   } else if (config.auth.plugin) {
-    //if it's something we don't recognize, try to require it by plugin name and pass in the options object
-    //if this provider wants to use the static client, it will have to call dbClientManager.useStaticClient()
-    //itself in the exported function.
-    try {
-      auth = require(config.auth.plugin)(config.auth.options);
-    }
-    catch(err) {
-      throw new Error(`auth plugin ${config.auth.plugin} not recognized`);
-    }
+    auth = require(config.auth.plugin)(config.auth.options);
   }
-
   if (auth) {
+    log.info(`Using auth plugin ${config.auth.plugin}`);
     koop.register(auth);
   }
-
-  log.info(`Using auth plugin ${config.auth.plugin}`);
-
 } else {
-  //if there's no auth provider configured, we have to use a direct pre-authenticated db client
   dbClientManager.useStaticClient(true);
-
   log.info(`No auth plugin specified, relying on configured MarkLogic credentials`);
 }
-// install the Marklogic Provider
+
+// Install the Marklogic Provider after installing an authorization plugin.
 const provider = require('./src/koop');
 koop.register(provider);
 
-// create the "global" app
+// Create and configure the Express app server.
 const app = express();
-
-if (config.enableServiceProxy) {
-  // proxy requests for Geo Data Services REST extensions and v1/documents
-  app.use(/\/(v1|LATEST)/,
-    proxy.create(/\/(resources\/(modelService|geoSearchService|geoQueryService)|documents)/));
-}
-log.info(`Service proxy for geo data services is ${(config.enableServiceProxy ? 'enabled' : 'disabled')}`);
-
-// otherwise route to Koop
+const port = config.port || 8080;
 app.use('/', koop.server);
+app.listen(port);
+console.log("Koop listening on port " + port + "; press ctrl-C to exit");
 
-// create HTTP server
-const server = http.createServer(app)
-  .listen(config.port || 80);
-log.info(`Koop MarkLogic Provider listening for HTTP on ${config.port}`);
-
-// also create an HTTPS server if enabled
-if (config.ssl.enabled) {
-  const https = require('https');
-  const options = {
-    key: fs.readFileSync(config.ssl.key),
-    cert: fs.readFileSync(config.ssl.cert)
-  };
-  https.createServer(options, app)
-    .listen(config.ssl.port || 443);
-  log.info(`Koop MarkLogic Provider listening for HTTPS on ${config.ssl.port}`);
-}
-
-console.log('Press control + c to exit');
-
-module.exports = server; // make it testable
+// module.exports = server; // make it testable
+// Disabling for now, need to understand what it would be used for.
+// This may just need to go into documentation.
+// if (config.enableServiceProxy) {
+//   const proxy = require('./src/koop/proxy');
+//   // proxy requests for Geo Data Services REST extensions and v1/documents
+//   app.use(/\/(v1|LATEST)/,
```

Comment:
This proxies the use of the MarkLogic REST API via koop. It is used by the ArcGIS plugin.